test/basics/test_linalg.jl

```diff
@@ -0,0 +1,35 @@
+using SparseArraysBase: SparseArrayDOK
+using LinearAlgebra: mul!
+using Random
```

Comment:
```suggestion
using Random: rand
```
Also probably we should set a seed, and use https://github.com/JuliaRandom/StableRNGs.jl to make sure the results are reproducible.